substrafl/strategies/fed_pca.py

```diff
@@ -0,0 +1,107 @@
+from typing import List
+from typing import Optional
+
+import numpy as np
+from numpy import linalg
+
+from substrafl.algorithms.algo import Algo
+from substrafl.exceptions import EmptySharedStatesError
+from substrafl.nodes.aggregation_node import AggregationNode
+from substrafl.nodes.references.local_state import LocalStateRef
+from substrafl.nodes.references.shared_state import SharedStateRef
+from substrafl.nodes.train_data_node import TrainDataNode
+from substrafl.remote import remote
+from substrafl.schemas import FedAvgAveragedState
+from substrafl.schemas import FedAvgSharedState
+from substrafl.schemas import StrategyName
+from substrafl.strategies import FedAvg
+
+
+class FedPCA(FedAvg):
+    """Federated Principal Component Analysis strategy."""
```

Comment:
The docstring will be converted and directly add to the user documentation. This is one of the most important part when adding new strategy, in order to explain the math behind it and to ensure the user what will append if he uses it. 
And this is also where your experience with the strategy will be more than appreciated.

As it have to be written in RST, and that it can be very tricky to obtain the result we expect, I propose you write it in Latex or md (or anything else) and send it to us. We will then convert it to rst to add it here.

Plz look at the FedAvg docstring to have an idea of what we can expect here.